### ISSUE TITLE
[clang-analyzer-security.insecureAPI.strcpy] extend list of unsafe functions

### ISSUE THREAD

**Issue by ingo-loehken** [llvm/llvm-project#103038]
Hi,

it would be nice, if the list of functions, that are marked as unsafe would be extensible or include i.e. the following (under windows and atl/mfc microsoft propietary stuff)
  - _strcpy
  - _strdup
  - _strcat
  - _tcscpy
  - _tcscat
  - _tcsdup
  - lstrcpy
  - lstrcat
  - strncpy
  - strncat

Check: https://clang.llvm.org/docs/analyzer/checkers.html#security-insecureapi-strcpy-c
Label : clang-tidy
Type: Enhancement

**Comment by steakhal** [comment]
Makes sense to me. We should doublecheck how those calls are named on Windows to be sure.

**Comment by haoNoQ** [comment]
Arguably `strncpy` is significantly "more" secure than `strcpy`. It's still insecure but it was specifically built to address the main problem with `strcpy`. So it shouldn't necessarily be on the same list just because the name sounds similar. It may be better to put it under a separate flag because folks may be ok with one but not the other.

**Comment by ParkHanbum** [comment]
Hi. I'd like to fix this issue, can you point me to a commit that I can refer to?

**Comment by fawdlstty** [comment]
@ParkHanbum Hi. Are you still working on it? I noticed that there has been no new information on this issue for a week in a row. Can you assign the issue to me? I would like to work on it and try and solve it, as a beginner coder. I will be very grateful to you.

**Comment by ParkHanbum** [comment]
@fawdlstty I didn't get this, so you don't have to ask me.

**Comment by fawdlstty** [comment]
I can't find the api that starts with `_` in my windows sdk (10.0.22621.0). Might it have been removed?

**Comment by ingo-loehken** [comment]
`<string.h>`, apis that start with `_`. But I did not find `_strcpy` yet...

**Comment by shrikardongre** [comment]
Hi  @fawdlstty , I also saw the other issue you mentioned , i was curious if you were still working on this  issue ,as a beginner to LLVM and clang I wanted to know if i can be of some help ,if you are still working on it .
If not can i get assigned to this issue so that i can learn more .

**Comment by fawdlstty** [comment]
@shrikardongre Hello, I'm sorry that I didn't continue this problem, you can try to solve it. Thank you for your dedication

**Comment by shrikardongre** [comment]
@llvmbot  can i be officialy assigned to this ?

**Comment by shrikardongre** [comment]
can i get assigned to this issue ?

**Comment by steakhal** [comment]
I don't think assigning would help make progress on solving the issue. I don't think anyone needs to know if someone want to work on an issue, or already started working on an issue. We would get notifications for new PRs mentioning this issue. That should be enough to "lock" an issue, and continue a discussion under that PR for a technical solution.

I speak I, and I don't think assigning issues in the Static Analyzer subproject fits into our workflow.

**Comment by shrikardongre** [comment]
Oh that's absolutely fine 
Thanks for clarification 😊
